File: examples-sdk/swift/001/main.swift
// 001 — AI Gateway (SSE Pipeline)
// Equivalent to: examples/001-basic-ai-gw-demo (Rust)
// Compile: vil compile --from swift --input 001/main.swift --release

// NOTE: Include SDK source from sdk/swift/Vil.swift when compiling.
// For single-file use: swift -I ../../sdk/swift/ main.swift

let pipeline = VilPipeline("ai-gateway", port: 3080)
    .sink("webhook", port: 3080, path: "/trigger")
    .source("inference", url: "http://localhost:4545/v1/chat/completions",
            format: "sse", jsonTap: "choices[0].delta.content", dialect: "openai")
    .route("webhook.trigger_out", to: "inference.trigger_in", mode: "LoanWrite")
    .route("inference.data_out", to: "webhook.data_in", mode: "LoanWrite")
    .route("inference.ctrl_out", to: "webhook.ctrl_in", mode: "Copy")

pipeline.compile()

File: examples-sdk/csharp/001/Main.cs
// 001 — AI Gateway (SSE Pipeline)
// Equivalent to: examples/001-basic-ai-gw-demo (Rust)
// Compile: vil compile --from csharp --input 001/Main.cs --release
#load "../../../sdk/csharp/Vil.cs"

var pipeline = new VilPipeline("ai-gateway", 3080)
    .Sink("webhook", 3080, "/trigger")
    .Source("inference", "http://localhost:4545/v1/chat/completions",
        format: "sse", dialect: "openai", jsonTap: "choices[0].delta.content")
    .Route("webhook.trigger_out", "inference.trigger_in", "LoanWrite")
    .Route("inference.data_out", "webhook.data_in", "LoanWrite")
    .Route("inference.ctrl_out", "webhook.ctrl_in", "Copy");

pipeline.Compile();
